src/kOS.Safe/Encapsulation/ListValue.cs

```diff
@@ -1,56 +1,119 @@
+        }
+
+        public int Count
+        {
+            get { return internalList.Count; }
+        }
+
+        public bool IsReadOnly
+        {
+            get { return internalList.IsReadOnly; }
+        }
+
+        public int IndexOf(T item)
+        {
+            return internalList.IndexOf(item);
+        }
+
+        public void Insert(int index, T item)
+        {
+            internalList.Insert(index, item);
+        }
+
+        public void RemoveAt(int index)
+        {
+            internalList.RemoveAt(index);
+        }
+
+        public T this[int index]
+        {
+            get { return internalList[index]; }
+            set { internalList[index] = value; }
         }
 
         private void ListInitializeSuffixes()
         {
-            AddSuffix("ADD",      new OneArgsSuffix<object>             (toAdd => list.Add(toAdd), Resources.ListAddDescription));
-            AddSuffix("INSERT",   new TwoArgsSuffix<int, int>           ((index,toAdd) => list.Insert(index,toAdd)));
-            AddSuffix("REMOVE",   new OneArgsSuffix<int>                (toRemove => list.RemoveAt(toRemove)));
-            AddSuffix("CLEAR",    new NoArgsSuffix                      (() => list.Clear()));
-            AddSuffix("LENGTH",   new NoArgsSuffix<int>                 (() => list.Count));
-            AddSuffix("ITERATOR", new NoArgsSuffix<Enumerator>          (() => new Enumerator(list.GetEnumerator())));
-            AddSuffix("COPY",     new NoArgsSuffix<ListValue>           (() => new ListValue(this)));
-            AddSuffix("CONTAINS", new OneArgsSuffix<bool, object>       (item => list.Contains(item)));
-            AddSuffix("SUBLIST",  new TwoArgsSuffix<ListValue, int, int>(SubListMethod));
-            AddSuffix("EMPTY",    new NoArgsSuffix<bool>                (() => !list.Any()));
-            AddSuffix("DUMP",     new NoArgsSuffix<string>              (ListDumpDeep));
+            AddSuffix("ADD", new OneArgsSuffix<T>(toAdd => internalList.Add(toAdd), Resources.ListAddDescription));
+            AddSuffix("INSERT", new TwoArgsSuffix<int, T>((index, toAdd) => internalList.Insert(index, toAdd)));
+            AddSuffix("REMOVE", new OneArgsSuffix<int>(toRemove => internalList.RemoveAt(toRemove)));
+            AddSuffix("CLEAR", new NoArgsSuffix(() => internalList.Clear()));
+            AddSuffix("LENGTH", new NoArgsSuffix<int>(() => internalList.Count));
+            AddSuffix("ITERATOR", new NoArgsSuffix<Enumerator>(() => new Enumerator(internalList.GetEnumerator())));
+            AddSuffix("COPY", new NoArgsSuffix<ListValue<T>>(() => new ListValue<T>(this)));
+            AddSuffix("CONTAINS", new OneArgsSuffix<bool, T>(item => internalList.Contains(item)));
+            AddSuffix("SUBLIST", new TwoArgsSuffix<ListValue, int, int>(SubListMethod));
+            AddSuffix("EMPTY", new NoArgsSuffix<bool>(() => !internalList.Any()));
+            AddSuffix("DUMP", new NoArgsSuffix<string>(ListDumpDeep));
```

Comment:
Sigh.  A lot of times automated re-indenting and re-spacing development tools actually get in the way of good spacing by preventing you from making stuff that lines up in columns.